tests/test_rcparams.py

```diff
@@ -183,11 +186,12 @@ def test_make_iterable_validator_none_auto(value, allow_auto, allow_none):
 def test_make_iterable_validator_length(value, length):
     scalar_validator = _validate_float_or_none
     validate_iterable = make_iterable_validator(scalar_validator, length=length)
+    assert validate_iterable
     raise_error = False
     if length is not None and len(value) != length:
-        raise_error = "Iterable must be of length"
+        raise_error = True
     if raise_error:
-        with pytest.raises(ValueError, match=raise_error):
+        with pytest.raises(ValueError, match="Iterable must be of length"):
```

Comment:
I have to say this was very dumb of mypy. Seeing the error I thought we were passing `match=True` which was clearly not the case.

Are there any alternatives to mypy? A lot of its inputs are useful, and we have caught many incorrect docstrings (which result in incorrect type hints) but this is also shaping up to be a significant effort now, some extra maintenance (not sure how much it will be but something) plus it makes it harder to work on the library. If there are less alternatives that are more pythonic it would be nice to consider them.